atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java

```diff
@@ -126,10 +113,29 @@ private CassandraClient instrumentClient(Client rawClient) {
 
     static CassandraClient getClientInternal(InetSocketAddress addr, CassandraKeyValueServiceConfig config)
             throws TException {
-        return new CassandraClientImpl(getRawClient(addr, config));
+        return new CassandraClientImpl(getRawClient(addr, config, createSslSocketFactory(config)));
     }
 
-    private static Cassandra.Client getRawClient(InetSocketAddress addr, CassandraKeyValueServiceConfig config)
+    private static SSLSocketFactory createSslSocketFactory(CassandraKeyValueServiceConfig config) {
+        return config.sslConfiguration().map(SslSocketFactories::createSslSocketFactory)
+                .orElseGet(() -> {
+                    try {
+                        /*
+                         * Use a separate SSLSocketFactory per host to reduce contention on the synchronized
```

Comment:
comment is out of date re per host, and is the comment body around nextBytes no longer relevant? haven't looked too closely at how `SSLSocketFactory` works.

Reply:
Yeah, it's now one across the cluster. I'm surprised this was a large source of allocations - @j-baker  would probably know better as he'd seen the data.